Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift

```diff
@@ -219,9 +223,10 @@ extension ExperienceStateMachine {
     enum SideEffect {
         case continuation(Action)
         case presentContainer(ExperienceData, Experience.StepIndex, ExperiencePackage, [Experience.Action])
+        case retryPresentation(ExperienceData, Experience.StepIndex, ExperiencePackage)
```

Comment:
this _could_ use the existing `.presentContainer`, but I preferred the new `.retryPresentation` to be explicit. It (1) avoids adding another `isRecovering` in the associated values of the existing enum value (which is a swiftlint violation) and (2) makes it clear on L252 below that we are in recovery mode, which is needed on L288 to not add duplicate pageMonitor observers.